### PR TITLE
Add "Rename" button to name-edit popover

### PR DESCRIPTION
**Pull request by ferdnyc**
In #610 I talked myself into thinking there should be one. :laughing: 

I also renamed the popover and all its children to `rename-popover`, `rename-entry`, etc... because "`service`..." both felt unintuitive to me, _and_ was used in other areas of the interface completely divorced from the popover, which felt _additionally_ confusing. I left the button in the headerbar named `service-edit`, though. Now it just brings up the `rename-popover`. 

Anyway, I can reverse that part of the change if you prefer.

### CODE REVIEW

**Comment by ferdnyc** [comment]
Cherry-picked my regexp fix from #615, to once again appease Travis.

File: data/ui/preferences-window.ui [hunk 1]
```diff
@@ -2,29 +2,83 @@
...
 <interface domain="org.gnome.Shell.Extensions.GSConnect">
   <requires lib="gtk+" version="3.20"/>
-  <object class="GtkPopover" id="service-popover">
+  <object class="GtkPopover" id="rename-popover">
     <property name="can_focus">False</property>
+    <signal name="focus-out-event" handler="_onUnfocusServiceName" swapped="no"/>
     <child>
-      <object class="GtkEntry" id="service-entry">
+      <object class="GtkGrid" id="rename">
+        <property name="name">rename</property>
```
Comment (andyholmes) [inline data/ui/preferences-window.ui]: I don't think `name` is necessary. It's really just some arbitrary text field we sometimes use to identify widgets, but can't really be used to find them like `id`. eg. `id` is what we use in the `Children: []` list.

File: data/ui/preferences-window.ui [hunk 2]
```diff
@@ -2,29 +2,83 @@
...
+        <property name="row_spacing">6</property>
+        <property name="column_spacing">6</property>
+        <child>
+          <object class="GtkLabel" id="rename-label">
+            <property name="visible">True</property>
+            <property name="can_focus">False</property>
+            <property name="halign">start</property>
+            <property name="margin_left">2</property>
+            <property name="margin_right">2</property>
+            <property name="margin_bottom">2</property>
```
Comment (andyholmes) [inline data/ui/preferences-window.ui]: No margins on the label itself; instead `column-spacing` and `row-spacing` of `6` on the parent `GtkGrid`.

File: data/ui/preferences-window.ui [hunk 3]
```diff
@@ -2,29 +2,83 @@
...
+            <property name="left_attach">1</property>
+            <property name="top_attach">1</property>
+          </packing>
+        </child>
       </object>
     </child>
   </object>
   <template class="GSConnectPreferencesWindow" parent="GtkApplicationWindow">
     <property name="can_focus">False</property>
-    <property name="icon_name">org.gnome.Shell.Extensions.GSConnect-symbolic</property>>
```
Comment (andyholmes) [inline data/ui/preferences-window.ui]: That's embarassing :/

**Comment by andyholmes** [comment]
As per my note on the translation thread, I'd like to get the translations updated sometime soon, so it'd be a good time to get this one in.

Did you want me to just make those nitpick changes myself?

**Comment by andyholmes** [comment]
Forked your MR and merged, so closing.

**Comment by ferdnyc** [comment]
@andyholmes Eep! Sorry for ignoring this one. It got completely buried in the deluge of Issue notification emails from a totally different project.

(Which I was supposed to be spending this weekend clearing out... and I think I did get through almost every message from before like Friday morning. And yet, my backlog is now **twice** as large now as it was then. \*sigh*)

**Comment by andyholmes** [comment]
No worries :P 

I'm in the same boat, trying to triage gnome-shell bugs and review others' extensions for the 3.34 update. I'm just glad I'm not one of those suckers maintaining GNOME Shell or something with *actually* a lot of users :laughing:

**Pull request closed without merging**